### PR TITLE
[FSDP] restricts meta model check to non ignored modules in FSDP

### PR DESCRIPTION
**Pull request by colin2328**
Summary: as title

Test Plan: see test plan D40287799

Differential Revision: D40287890

### CODE REVIEW

**Review by awgu — APPROVED** [review]
LGTM!

**Review by rohan-varma — COMMENTED** [review]
We should ideally include an OSS unit test that would've caught this issue, otherwise LGTM. Thanks for the fix!

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -1367,7 +1368,7 @@ def _materialize_module(
         ``reset_parameters()``, and for torchdistX fake tensors, this calls
         ``deferred_init.materialize_module()``.
         """
-        is_meta_module = any(p.is_meta for p in module.parameters())
+        is_meta_module = any(p.is_meta for p in self._get_orig_params(module, ignored_params))
```
Comment (rohan-varma) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:1371]: nit: the fake.is_fake check on 1375 may also have to be updated. But feel free to file an issue if not in scope of this PR.

**Comment by rohan-varma** [comment]
@colin2328 @awgu Could we get a UT added here + follow up on the fake tensor thing as well

**Pull request closed without merging**